From: Christian Hofstaedtler <zeha@debian.org>
Subject: Switch to B_ASN1_UTF8STRING

Only use UTF8 encoding in ASN.1 strings.
This is recommended by RFC2459 (2004), and has been made the default in
OpenSSL 1.0.1h. Fixes the FTBFS caused by the OpenSSL change.

diff --git a/src/racoon/eaytest.c b/src/racoon/eaytest.c
index 1474bdc..b1a3bbf 100644
--- a/src/racoon/eaytest.c
+++ b/src/racoon/eaytest.c
@@ -266,17 +266,17 @@ certtest(ac, av)
 		0x30,0x7a,0x31,0x0b,0x30,0x09,0x06,0x03,
 		0x55,0x04,0x06,0x13,0x02,0x4a,0x50,0x31,
 		0x11,0x30,0x0f,0x06,0x03,0x55,0x04,0x08,
-		0x13,0x08,0x4b,0x61,0x6e,0x61,0x67,0x61,
+		0x0c,0x08,0x4b,0x61,0x6e,0x61,0x67,0x61,
 		0x77,0x61,0x31,0x11,0x30,0x0f,0x06,0x03,
-		0x55,0x04,0x07,0x13,0x08,0x46,0x75,0x6a,
+		0x55,0x04,0x07,0x0c,0x08,0x46,0x75,0x6a,
 		0x69,0x73,0x61,0x77,0x61,0x31,0x15,0x30,
-		0x13,0x06,0x03,0x55,0x04,0x0a,0x13,0x0c,
+		0x13,0x06,0x03,0x55,0x04,0x0a,0x0c,0x0c,
 		0x57,0x49,0x44,0x45,0x20,0x50,0x72,0x6f,
 		0x6a,0x65,0x63,0x74,0x31,0x15,0x30,0x13,
-		0x06,0x03,0x55,0x04,0x0b,0x13,0x0c,0x4b,
+		0x06,0x03,0x55,0x04,0x0b,0x0c,0x0c,0x4b,
 		0x41,0x4d,0x45,0x20,0x50,0x72,0x6f,0x6a,
 		0x65,0x63,0x74,0x31,0x17,0x30,0x15,0x06,
-		0x03,0x55,0x04,0x03,0x13,0x0e,0x53,0x68,
+		0x03,0x55,0x04,0x03,0x0c,0x0e,0x53,0x68,
 		0x6f,0x69,0x63,0x68,0x69,0x20,0x53,0x61,
 		0x6b,0x61,0x6e,0x65,
 	};
diff --git a/src/racoon/crypto_openssl.c b/src/racoon/crypto_openssl.c
index c192904..7d97192 100644
--- a/src/racoon/crypto_openssl.c
+++ b/src/racoon/crypto_openssl.c
@@ -147,6 +147,8 @@ eay_str2asn1dn(str, len)
 	}
 	memcpy(buf, str, len);
 
+	/* Set RFC2459 recommended mode, default in OpenSSL 1.0.1h+ */
+	ASN1_STRING_set_default_mask(B_ASN1_UTF8STRING);
 	name = X509_NAME_new();
 
 	dst = field = &buf[0];
